File: smartlist/src/main/java/com/smartlist/module/report/dto/response/AssessmentBlockCandidatePerformanceRespDTO.java
package com.smartlist.module.report.dto.response;

import com.fasterxml.jackson.annotation.JsonProperty;
import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Positive;
import jakarta.validation.constraints.PositiveOrZero;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class AssessmentBlockCandidatePerformanceRespDTO {

    @Schema(description = "Candidate Id", example = "", requiredMode= Schema.RequiredMode.REQUIRED)
    @NotNull
    @Positive
    @JsonProperty(value = "candidate_id")
    Integer candidateId;

    @Schema(description = "Candidate Name", example = "", requiredMode= Schema.RequiredMode.REQUIRED)
    @NotEmpty
    @JsonProperty(value = "candidate_name")
    String candidateName;

    @Schema(description = "Score", example = "", requiredMode= Schema.RequiredMode.REQUIRED)
    @NotNull
    @PositiveOrZero
    Long score;


}

File: javascript/smartlist/smartlist-api-mysql-project/src/modules/report/dto/response/assessment-block-candidate-performance.resp.dto.ts
import { ApiProperty } from '@nestjs/swagger';

export class AssessmentBlockCandidatePerformanceRespDto {
  @ApiProperty()
  id: number;

  @ApiProperty()
  name: string;

  @ApiProperty()
  score: number;

  constructor(rawData: any) {
    this.id = Number(rawData.id);
    this.name = rawData.name;
    this.score = Number(rawData.score);
  }
}
